### DISCUSSION TITLE
Best practice for uploading and indexing a large dataset

### DISCUSSION THREAD

**Discussion by dsp05** [milvus-io/milvus#22637] [Q&A and General discussion] [answered]
Hi team, it looks like there are two ways to update data when creating a new collection:
- upload the data, then call `create_index`
- call `create_index` on the empty collection, then upload the data.

If the dataset is large, e.g. >100GB, is there any difference between the two, and which one is recommended?

Thanks!

**Comment by yhmo** [comment, marked as answer]
- upload the data, then call create_index()
  During you inserting data into milvus, the index nodes are idle. the index nodes will start building index after you called create_index().
  This approach is suitable for small and non-streaming dataset.

- call create_index() on the empty collection, then upload the data
  During you inserting data into milvus, the data nodes receive data and accumulate data in a buffer. Once the buffer is full, datanode will flush the buffer into storage and generated a new segment. The index nodes can detect new segments and automatically build index for new segments. So, insert data operations and index progress are executed parallelly, which will save your time.
  This approach is suitable for large and streaming dataset.
